src/convert.cpp

```diff
@@ -1567,7 +1573,7 @@ bool convertStringCharsetIconv(std::string& str, const char* from, const char* t
   bool ret = true;
   iconv_t cd;
   cd = iconv_open(to, from);
-  if (!cd) {
+  if (cd == (iconv_t)(-1)) {
```

Comment:
[suggestion] could you define and assign the `cd` in the same line? Furthermore, is it really needed the cast to `iconv_t` if we already know that the variable `cd` is of that type?

Reply:
musl libc (and most others) have it as
typedef void *iconv_t;

uclibc-ng has it as
typedef long iconv_t;

cast is needed if it's a pointer I think.

This does trigger a clang-tidy check that it's a C cast.

I wonder if there's a C++ way to do this...

Reply:
@neheb
> musl libc (and most others) have it as
> typedef void *iconv_t;
>
> uclibc-ng has it as
> typedef long iconv_t;

Thanks for letting me know, I wasn't aware of the different definitions. As for a C++ solution, would using `std::is_pointer_v` help?

Reply:
No i mean you can't use static_cast or reinterpret_cast as it would fail with void* and long respectively.

```
if std::is_pointer_v<cd> ? reinterpret_cast<iconv_t>(-1) : static_cast<iconv_t>(-1))
```

is quite verbose.

```
if (iconv_t(-1))
```

is an alternative but newer clang-tidy warns on that as well.